Editor/EditorCore/BezierMeshEditor.cs

```diff
@@ -0,0 +1,61 @@
+#if USING_SPLINES && UNITY_2021_3_OR_NEWER
+
+using System.Collections.Generic;
+using UnityEditor.EditorTools;
+using UnityEditor.Splines;
+using UnityEngine;
+using UnityEngine.ProBuilder;
+
+namespace UnityEditor.ProBuilder
+{
+    [CustomEditor(typeof(BezierMesh))]
+    [CanEditMultipleObjects]
+    class BezierMeshEditor : Editor
+    {
+        BezierMeshOverlay m_Overlay;
+        List<BezierMesh> m_SelectedMeshes;
+        bool m_isOverlayVisible;
+
+        void BuildSelection()
+        {
+            m_SelectedMeshes = new List<BezierMesh>();
+
+            foreach (var obj in Selection.gameObjects)
+            {
+                if (obj.TryGetComponent(out BezierMesh mesh))
+                {
+                    m_SelectedMeshes.Add(mesh);
+                }
+            }
+
+            m_isOverlayVisible = m_SelectedMeshes.Count > 0;
+        }
+
+        void OnEnable()
+        {
+            BuildSelection();
+#if UNITY_2022_1_OR_NEWER
+            SceneView.AddOverlayToActiveView(m_Overlay = new BezierMeshOverlay(m_SelectedMeshes, m_isOverlayVisible));
+#endif
+            BezierMesh.BezierMeshModified += () => ProBuilderEditor.Refresh();
+        }
+
+        void OnDisable()
+        {
+#if UNITY_2022_1_OR_NEWER
+            SceneView.RemoveOverlayFromActiveView(m_Overlay);
+#endif
+            BezierMesh.BezierMeshModified -= () => ProBuilderEditor.Refresh();
+        }
```

Comment:
Because you're registering a lambda expression anonymous function to `BezierMesh.BezierMeshModified` it will not unregister during`OnDisable`. Basically, every time you just do `() => ProBuilderEditor.Refresh()` you're implicitly defining a new delegate. So when you do `BezierMesh.BezierMeshModified -= () => ProBuilderEditor.Refresh();` you're basically trying to deregister a delegate you've just created. Here's some info on this: https://stackoverflow.com/questions/183367/unsubscribe-anonymous-method-in-c-sharp
You can test that the ` BezierMesh.BezierMeshModified` accumulates registrations by adding a debug log as part of the callback and then duplicate an object many times, then leave just one instance and duplicate it again - you'll see many debug outputs.